src/binding/petsc4py/src/petsc4py/PETSc/DMDA.pyx

```diff
@@ -422,27 +1008,90 @@ cdef class DMDA(DM):
                                       &refine[2]) )
         return tuple([toInt(refine[i]) for 0 <= i < dim])
 
-    def setInterpolationType(self, interp_type):
+    def setInterpolationType(self, interp_type: InterpolationType) -> None:
+        """Set the type of interpolation.
+
+        You should call this on the coarser of the two DMDAs you pass to
+        `DM.createInterpolation`.
+
+        Logically collective.
+
+        Parameters
+        ----------
+        interp_type
+            The interpolation type.
+
+        See Also
+        --------
+        getInterpolationType, petsc.DMDASetInterpolationType
+
+        """
         cdef PetscDMDAInterpolationType ival = dainterpolationtype(interp_type)
         CHKERR( DMDASetInterpolationType(self.dm, ival) )
 
-    def getInterpolationType(self):
+    def getInterpolationType(self) -> int:
+        """Return the type of interpolation.
+
+        Not collective.
+
+        See Also
+        --------
+        setInterpolationType, petsc.DMDAGetInterpolationType
+
+        """
         cdef PetscDMDAInterpolationType ival = DMDA_INTERPOLATION_Q0
         CHKERR( DMDAGetInterpolationType(self.dm, &ival) )
         return <long>ival
```

Comment:
Wow, this should be fixed. Ask @dalcinl

Reply:
He saw it and did not like it but managed to stopped himself from fixing it just yet. I'll add a FIXME.